app/bundles/EmailBundle/Helper/MailHelper.php

```diff
@@ -1674,7 +1674,13 @@ public function createEmailStat($persist = true, $emailAddress = null, $listId =
         $emailExists = $this->email && $this->email->getId();
 
         if ($emailExists) {
-            $stat->setEmail($this->email);
+            $reference = $this->entityManager->getReference(Email::class, (int) $this->email->getId());
+            if (!$reference) {
+                $this->logger->warning('Email does not exist in database.', [
+                    'emailId' => $this->email->getId(),
+                ]);
+            }
```

Comment:
I think the `if` statement doesn't need to be there. When you look at what the `getReference` method does:

> Gets a reference to the entity identified by the given type and identifier without actually loading it, if the entity is not yet loaded.

So it's not checking the database. Just creates the entity with the ID if not already present in EntityManager.
```suggestion
            $reference = $this->entityManager->getReference(Email::class, (int) $this->email->getId());
```

Reply:
Im fine with that change. I added it as a fallback as there was a test case failing, it returned null. But I guess that was then only a problem with the test scenario.